source/fab/cli.py

```diff
@@ -12,28 +12,25 @@
 from fab.steps.compile_c import compile_c
 from fab.steps.link import link_exe
 from fab.steps.root_inc_files import root_inc_files
-
-import fab
 from fab.artefacts import CollectionGetter
 from fab.build_config import BuildConfig
 from fab.constants import PRAGMAD_C
 from fab.steps.compile_fortran import compile_fortran, get_fortran_compiler
 from fab.steps.find_source_files import find_source_files
 from fab.steps.grab.folder import grab_folder
 from fab.steps.preprocess import preprocess_c, preprocess_fortran
+from fab.util import common_arg_parser
 
 
-def _generic_build_config(folder: Path, kwargs: Optional[Dict] = None) -> BuildConfig:
-    folder = folder.resolve()
-    kwargs = kwargs or {}
+def _generic_build_config(folder: str = '.', kwargs=None) -> BuildConfig:
```

Comment:
What was the reason for falling back to path-as-string? If we can keep them as Path objects we should. The default constructor for Path gives the currently selected directory (e.g. `Path()`) if that was the problem.

Reply:
This is only used to create the 'project_label' when using 'zero config' fab, so it simplifies the process.

Reply:
My mistake it should have been a Path object